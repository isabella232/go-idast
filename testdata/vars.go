package vars

var A = 1
var b = A + 2
var c = "foo"
