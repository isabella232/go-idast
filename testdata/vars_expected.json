 Package         | (n/a)                           | Package
 File            | package vars\n\nvar A = 1\nvar  | Package/Files/testdata/vars.go
 Ident           | vars                            | Package/Files/testdata/vars.go/Name/Ident
 GenDecl         | var A = 1                       | Package/Files/testdata/vars.go/Decls/0/GenDecl
 ValueSpec       | A = 1                           | Package/Files/testdata/vars.go/Decls/0/GenDecl/Specs/0/ValueSpec
 Ident           | A                               | Package/Files/testdata/vars.go/Decls/0/GenDecl/Specs/0/ValueSpec/Names/A/Ident
 BasicLit        | 1                               | Package/Files/testdata/vars.go/Decls/0/GenDecl/Specs/0/ValueSpec/Values/0/BasicLit
 GenDecl         | var b = A + 2                   | Package/Files/testdata/vars.go/Decls/1/GenDecl
 ValueSpec       | b = A + 2                       | Package/Files/testdata/vars.go/Decls/1/GenDecl/Specs/0/ValueSpec
 Ident           | b                               | Package/Files/testdata/vars.go/Decls/1/GenDecl/Specs/0/ValueSpec/Names/b/Ident
 BinaryExpr      | A + 2                           | Package/Files/testdata/vars.go/Decls/1/GenDecl/Specs/0/ValueSpec/Values/0/BinaryExpr
 Ident           | A                               | Package/Files/testdata/vars.go/Decls/1/GenDecl/Specs/0/ValueSpec/Values/0/BinaryExpr/X/Ident
 BasicLit        | 2                               | Package/Files/testdata/vars.go/Decls/1/GenDecl/Specs/0/ValueSpec/Values/0/BinaryExpr/Y/BasicLit
 GenDecl         | var c = "foo"                   | Package/Files/testdata/vars.go/Decls/2/GenDecl
 ValueSpec       | c = "foo"                       | Package/Files/testdata/vars.go/Decls/2/GenDecl/Specs/0/ValueSpec
 Ident           | c                               | Package/Files/testdata/vars.go/Decls/2/GenDecl/Specs/0/ValueSpec/Names/c/Ident
 BasicLit        | "foo"                           | Package/Files/testdata/vars.go/Decls/2/GenDecl/Specs/0/ValueSpec/Values/0/BasicLit
 File            | package vars\n\nfunc A(b, c str | Package/Files/testdata/funcs.go
 Ident           | vars                            | Package/Files/testdata/funcs.go/Name/Ident
 FuncDecl        | func A(b, c string, d int) (w,  | Package/Files/testdata/funcs.go/Decls/0/FuncDecl:A
 Ident           | A                               | Package/Files/testdata/funcs.go/Decls/0/FuncDecl:A/Name/Ident
 FuncType        | func(b, c string, d int) (w, x  | Package/Files/testdata/funcs.go/Decls/0/FuncDecl:A/Type/FuncType
 FieldList       | (n/a)                           | Package/Files/testdata/funcs.go/Decls/0/FuncDecl:A/Type/FuncType/Params/FieldList
 Field           | (n/a)                           | Package/Files/testdata/funcs.go/Decls/0/FuncDecl:A/Type/FuncType/Params/FieldList/List/0/Field
 Ident           | b                               | Package/Files/testdata/funcs.go/Decls/0/FuncDecl:A/Type/FuncType/Params/FieldList/List/0/Field/Names/b/Ident
 Ident           | c                               | Package/Files/testdata/funcs.go/Decls/0/FuncDecl:A/Type/FuncType/Params/FieldList/List/0/Field/Names/c/Ident
 Ident           | string                          | Package/Files/testdata/funcs.go/Decls/0/FuncDecl:A/Type/FuncType/Params/FieldList/List/0/Field/Type/Ident
 Field           | (n/a)                           | Package/Files/testdata/funcs.go/Decls/0/FuncDecl:A/Type/FuncType/Params/FieldList/List/1/Field
 Ident           | d                               | Package/Files/testdata/funcs.go/Decls/0/FuncDecl:A/Type/FuncType/Params/FieldList/List/1/Field/Names/d/Ident
 Ident           | int                             | Package/Files/testdata/funcs.go/Decls/0/FuncDecl:A/Type/FuncType/Params/FieldList/List/1/Field/Type/Ident
 FieldList       | (n/a)                           | Package/Files/testdata/funcs.go/Decls/0/FuncDecl:A/Type/FuncType/Results/FieldList
 Field           | (n/a)                           | Package/Files/testdata/funcs.go/Decls/0/FuncDecl:A/Type/FuncType/Results/FieldList/List/0/Field
 Ident           | w                               | Package/Files/testdata/funcs.go/Decls/0/FuncDecl:A/Type/FuncType/Results/FieldList/List/0/Field/Names/w/Ident
 Ident           | x                               | Package/Files/testdata/funcs.go/Decls/0/FuncDecl:A/Type/FuncType/Results/FieldList/List/0/Field/Names/x/Ident
 Ident           | float                           | Package/Files/testdata/funcs.go/Decls/0/FuncDecl:A/Type/FuncType/Results/FieldList/List/0/Field/Type/Ident
 Field           | (n/a)                           | Package/Files/testdata/funcs.go/Decls/0/FuncDecl:A/Type/FuncType/Results/FieldList/List/1/Field
 Ident           | y                               | Package/Files/testdata/funcs.go/Decls/0/FuncDecl:A/Type/FuncType/Results/FieldList/List/1/Field/Names/y/Ident
 Ident           | bool                            | Package/Files/testdata/funcs.go/Decls/0/FuncDecl:A/Type/FuncType/Results/FieldList/List/1/Field/Type/Ident
 BlockStmt       | {\n	return len(b+c) + d\n}      | Package/Files/testdata/funcs.go/Decls/0/FuncDecl:A/Body/BlockStmt
 ReturnStmt      | return len(b+c) + d             | Package/Files/testdata/funcs.go/Decls/0/FuncDecl:A/Body/BlockStmt/List/0/ReturnStmt
 BinaryExpr      | len(b+c) + d                    | Package/Files/testdata/funcs.go/Decls/0/FuncDecl:A/Body/BlockStmt/List/0/ReturnStmt/Results/0/BinaryExpr
 CallExpr        | len(b + c)                      | Package/Files/testdata/funcs.go/Decls/0/FuncDecl:A/Body/BlockStmt/List/0/ReturnStmt/Results/0/BinaryExpr/X/CallExpr
 Ident           | len                             | Package/Files/testdata/funcs.go/Decls/0/FuncDecl:A/Body/BlockStmt/List/0/ReturnStmt/Results/0/BinaryExpr/X/CallExpr/Fun/Ident
 BinaryExpr      | b + c                           | Package/Files/testdata/funcs.go/Decls/0/FuncDecl:A/Body/BlockStmt/List/0/ReturnStmt/Results/0/BinaryExpr/X/CallExpr/Args/0/BinaryExpr
 Ident           | b                               | Package/Files/testdata/funcs.go/Decls/0/FuncDecl:A/Body/BlockStmt/List/0/ReturnStmt/Results/0/BinaryExpr/X/CallExpr/Args/0/BinaryExpr/X/Ident
 Ident           | c                               | Package/Files/testdata/funcs.go/Decls/0/FuncDecl:A/Body/BlockStmt/List/0/ReturnStmt/Results/0/BinaryExpr/X/CallExpr/Args/0/BinaryExpr/Y/Ident
 Ident           | d                               | Package/Files/testdata/funcs.go/Decls/0/FuncDecl:A/Body/BlockStmt/List/0/ReturnStmt/Results/0/BinaryExpr/Y/Ident
